src/app/chat/view.nim

```diff
@@ -28,16 +33,34 @@ QtObject:
     for msg in self.messageList.values:
       msg.delete
     self.messageList = initTable[string, ChatMessageList]()
+    self.mStickers = initTable[int, StickerList]()
     self.QAbstractListModel.delete
 
   proc newChatsView*(status: Status): ChatsView =
     new(result, delete)
     result.status = status
     result.chats = newChannelsList()
     result.activeChannel = newChatItemView(status)
+    result.activeStickerPackId = -1
     result.messageList = initTable[string, ChatMessageList]()
+    result.mStickerPacks = newStickerPackList()
+    result.mStickers = initTable[int, StickerList]()
+    result.emptyStickerList = newStickerList()
     result.setup()
 
+  proc addStickerPackToList*(self: ChatsView, stickerPack: StickerPack) =
+    discard self.mStickerPacks.addStickerPackToList(stickerPack)
+    let stickerList = newStickerList()
+    for sticker in stickerPack.stickers:
+      discard stickerList.addStickerToList(sticker)
+    self.mStickers[stickerPack.id] = stickerList
```

Comment:
these 4 lines can be simplified with:

```nim
self.mStickers[stickerPack.id] = newStickerList(stickerPack.stickers)
```

with the constructor being modified to receive a the sticker seq

Reply:
👍